### ISSUE TITLE
Obfuscation of error during parsing with custom converter is surprising behavior

### ISSUE THREAD

**Issue by colindean** [scallop/scallop#224]
I've got a custom ValueConverter using `listArgConverter`. Inside that converter, there's a method outside my control that can throw an exception. Specifically, I'm parsing a URI and the JVM defines what URIs it can handle. Things are fine when I pass an argument that includes a URI that the JVM knows, but when I pass something it doesn't handle, Scallop exits, giving me an error with "wrong arguments format". It took way too long to realize that Scallop 3.4.0 is eating the exception that occurs:

https://github.com/scallop/scallop/blob/8d03638b05a0cf32cc0753c2c7073a107c099417/src/main/scala/org.rogach.scallop/DefaultConverters.scala#L83-L85

This is a bit surprising and unfortunately took an uncomfortable amount of time to track down.

**Comment by colindean** [comment]
It looks like I'm behind the times, but not by much: I see now that [Scallop 4.0.0](https://github.com/scallop/scallop/releases/tag/v4.0.0) came out a ~month ago and [alters this behavior](https://github.com/scallop/scallop/commit/9693fd07d9e976482f57721c1db006f165d133f8#diff-87c2bf754fb59a7b611edb2df506dd9b0c88f12134e76ba619a4cd3d402c30b6):

https://github.com/scallop/scallop/blob/9191d86119720c2ab9503b55ad9cfee48d0c0015/src/main/scala/org.rogach.scallop/DefaultConverters.scala#L92-L94

**Comment by colindean** [comment]
Interestingly enough, after upgrading to Scallop 4.0.1 an `ExcessArguments` exception is now obfuscating the underlying exception. I speculate that code path now causes the unprocessed argument to be treated as a trailing option, but without a trailing option ScallopOpt in place, the parser fails and doesn't ever have the chance to emit the converter exception.

Fortunately, I was able to get to that underlying exception much more easily having upgraded, so I can more easily fix it!

**Comment by Rogach** [comment]
Can you provide a full example for `ExcessArguments` problem?

**Comment by colindean** [comment]
Yes! This might be a little _too_ much to demonstrate but I took what I had and pared it down: https://scastie.scala-lang.org/MOJN9X4ZSoaZQyOxZrmILg

```
org.rogach.scallop.exceptions.ExcessArguments: Excess arguments provided: 'hdfs:///something/somewhere'
	at org.rogach.scallop.Scallop.parseRest$1(Scallop.scala:111)
	at org.rogach.scallop.Scallop.goParseRest$1(Scallop.scala:151)
	at org.rogach.scallop.Scallop.parse(Scallop.scala:197)
	at org.rogach.scallop.Scallop.parse(Scallop.scala:85)
	at org.rogach.scallop.Scallop.org$rogach$scallop$Scallop$$parsed$lzycompute(Scallop.scala:291)
	at org.rogach.scallop.Scallop.org$rogach$scallop$Scallop$$parsed(Scallop.scala:291)
	at org.rogach.scallop.Scallop.verify(Scallop.scala:515)
	at org.rogach.scallop.ScallopConfBase.verifyBuilder(ScallopConfBase.scala:626)
	at org.rogach.scallop.ScallopConfBase.verify(ScallopConfBase.scala:1087)
	at Playground$CliParams.<init>(main.scala:19)
```

**Comment by Rogach** [comment]
Here's a smaller example of the problem:

```scala
object Conf extends ScallopConf(Seq("-a", "1", "x")) {
  val apples = opt[List[Int]]()
  verify()
}
Conf.apples() shouldBe List(1)
```

The issue is due to our multiple-list parsing support (when we have list option followed by trailing args, or multiple list trailing args). In general when parser sees list options (or trailing args) it has to rely on converter errors to figure out the list boundaries.

In this specific case the solution is probably simple - if there is only one list option (or only one trailing argument) we should ditch excess arguments exception altogether and pass on the error from the converter.

I don't see a solution for general case, it's probably more like "avoid multiple trailing arg lists if you want better error messages".

**Comment by Rogach** [comment]
Published a snapshot with this fix:
```
resolvers += "Sonatype OSS Snapshots" at "https://oss.sonatype.org/content/repositories/snapshots"
libraryDependencies += "org.rogach" %% "scallop" % "4.0.2-SNAPSHOT"
```

**Comment by Rogach** [comment]
Released in version `4.0.2`.
